src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,72 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { currentTodoSlice } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [user, setUser] = useState<User | null>(null);
+
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const dispatch = useAppDispatch();
+
+  function handleCloseModal() {
+    dispatch(currentTodoSlice.actions.changeCurrentTodo(null));
+  }
+
+  useEffect(() => {
+    setIsLoading(true);
+    getUser(currentTodo!.userId)
+      .then(setUser)
+      .catch(() => {
+        throw new Error('User load fail');
```

Comment:
Throwing an error in the catch block will terminate the promise chain and potentially crash the application. Consider handling the error more gracefully, such as displaying an error message to the user.